### DISCUSSION TITLE
Best Practices  & Discussion for new instance admins

### DISCUSSION THREAD

**Discussion by ghost** [mastodon/mastodon#20002] [General]
Hi,

testing mastodon with a single user instance so far and playing around with the possibilites and issues. 
Like many others in the last days, as it seems. My goal was to build a public instance over time, but I see some problems I have to solve so far and which would be very complicated to explain to potential users.

As more as I read here on github and discussions, I get a better feeling about the technical backgrounds around those issues,  which I realy can understand. 
But the more I read, the more I get the feeling - that there is no good starting point for new instance admins, for example:
- "what are valid reasons to host your own instance and what is not"
- "what are the issues, and how could you improve" 
- "how to handle the bad user experience in the first time, untill your instance reaches a wider population".

Maybe I am wrong and just haven't found a good overview about this. In this case I would be nice, if someone could point that out for me.
If not - wouldn't it be a good idea to give it a central overview about this in the mastodon documentation or somewhere else?

One of the issues I run into is: If I want to spread my own instance and open it for the public - how do I deal with the empty timelines, inconsitent hashtag searches compared to other instance etc.
 What new users will see in the first time one a small but growing instance is in no way the same compared what they would see on other instances. For example like the issue explained here: https://github.com/mastodon/mastodon/issues/14017

I am following hastags on some accounts I have on bigger plattforms and I follow them on my own instance. And the one on my instance is not nearly in sync with the others. I tried to follow more and more people on those hashtags, and I tried crawlers like mastodon-hashtag-spreader for example. But it is far from perfect and I think crawlers would not be the desired solution for other instance admins because it would produce huge anoying network traffic.
Also "following bots" I guess would be discussed controversial.

The more and more mastodon will grow - the more this will be an issue. Maybe single user instances are a bad idea. But every plattform will start from that point. How do other admins handle this. 
Would be glad to see some testimonials about this.

**Comment by redstrate** [comment]
I wrote a little bit about single-user instances [in a blog post](https://redstrate.com/blog/2022/11/thoughts-and-tips-on-mastodon/), but I'll post the relevant excerpt here:

> I summarized this point in some random thread on Knockout, but wanted to explain it better here. In most cases, which is strange - I recommend not doing so. Here’s why:
> 
> Mastodon federation is not retroactive.
> 
> This might not seem immediately obvious, but here’s how it works: You spin up a new ActivityPub-compatible server (Mastodon, Pleroma, or other) and create a new account handle. Federation works as follows: you must interact with a server first before it starts talking to you. The easiest way to get started is to start following accounts. Once you do, the server starts to open up to you and sending it’s timeline, users, and posts your way. The more servers you interact with, the bigger your federated timeline becomes and more federated users are “known” to your server.
> 
> Unfortunately, this is where the problem lies therein - if you are spinning up your own Mastodon server just for yourself - it’s just not enough. If you want to see someone’s profile locally, a lot of their posts will be missing, because most likely they created their account way before your server even existed. In an unfortunate twist, the longer a Mastodon server existed the more likely it will have “more” content, and more complete user profiles available. For Pleroma users especially, it’s hard to interact with content that you are not initially federated to, because it lacks “reply gateways”, something that Mastodon users can take advantage of.
> 
> Also as a single user Mastodon instance, it’s going to be really hard to get going because the easiest way to start following accounts is by looking at the local and federated timelines, something that will be very barren or maybe non-existent for new servers:
> 
> Mastodon in general is designed (perhaps intentionally or not) for communities to form around multiple servers, and something like a single-user Mastodon instance is niche, and nigh unusable (in my opinion, as someone who has tried to run one before). In this way, it actually differs from hosting your own email.

Basically (right now) it's pretty tough to get a new federated instance hosting, moreso if you're the only user actually trying to interact. It's not _impossible_, but it's pretty lonely at first. I hear there's some ongoing work to make it so hashtags synchronize better between servers though.

**Reply by ghost** [reply]
(rewrite of my last comment to make it more precise):

Thanks for the good explanation.
As I understand - more or less it is mainly an issue about the activitypub protocoll itself. Or about missing tools or services around this.

Never the less: Mastodon is the application which got the most attention right now.
Everytime something goes wrong with twitter etc., lots of users try to switch and lots of "newbie" admins will play with creating instances.

If you visit any instance the first time and you want to register, you will get a banner which tells you "and you can easily host it yourself".
(do not have the english translation right now. In german it is "Sie können ihn sogar selber hosten". 
No hint about the drawbacks. This is somehow irritating!

To prevent disappointed feedback and a bad user/admin experience I think it would be a good to give a more visible and explicit hint in the
documentation and the banner etc. Maybe give a link to some platform which collects those topics.

It should be made more clear that every small instance could be contraproductive for the community itself. Because This confussion could lead to a bad reputation about the platform, the more and more instances will spawn up.

For me that has not been clear in the beginning. As a sysadmin my first reflex always is to host by myself.

I realy love the way the fediverse works. And IMHO it's realy worth to pay much attention how it could develop in terms of acceptance and to prevent those traps in the future.
